Type 'undefined' cannot be used as an index type

FrankStappers opened the pull request:
When installing/upgrading yarn (yarn install or yarn upgrade) and afterwards executing "yarn run webpack", the process fails:

```
ERROR in /projects/akeneo-pim/web/bundles/pimui/js/view/base.ts
./web/bundles/pimui/js/view/base.ts
[tsl] ERROR in /path/to/project/akeneo-pim/web/bundles/pimui/js/view/base.ts(94,28)
      TS2538: Type 'undefined' cannot be used as an index type.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The problem affects Akeneo "tags/v3.2.65" (and most probably also other versions)

<!--- (<3 Thanks for taking the time to contribute! You're awesome! <3) --->

<!--- (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md) --->

**Description (for Contributor and Core Developer)**

<!--- (What does this Pull Request do? reference the related issue?) --->
Fixes type 'undefined' index type usage.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

nestorAkeneo commented:
:heart_eyes: Thanks for taking the time to contribute! You're awesome! :heart_eyes:

In order for your contribution to benefit all other contributors and users, we need your permission.

For that, it is enough that you read this document, and that you sign it, if it suits you.

[Contributor Agreement](http://www.akeneo.com/contributor-license-agreement/)

jmleroux commented:
Fixed here with underscoreJS dependency: https://github.com/akeneo/pim-community-standard/pull/1373/files

FrankStappers commented:
Hi @jmleroux 

Thanks for the response. I have upgraded to Akeneo v3.2.66, which contains the aforementioned patch.
When I run the commands again, the problem still persists.

jmleroux commented:
Did you update your node_modules?

The simplest way to do it is to delete yarn.lock and nodes_modules and launch a `yarn install`

FrankStappers commented:
Hi @jmleroux,

I've removed the node_modules directory, removed the yarn.lock, and updated node to v14.7.0 (previous version v10.15.0).
When running `yarn install`, I still run into the `Type 'undefined' cannot be used as an index type` error.

Olaf1989 commented:
@jmleroux I can confirm that the problem is occurring (also see linked issue).
